.github/workflows/haskell.yml

```diff
@@ -40,14 +40,14 @@ jobs:
     - name: Install dependencies
       run: |
         cabal v2-update
-        cabal v2-configure --enable-benchmarks --enable-tests --write-ghc-environment-files=always -j2
-        cabal v2-build --only-dependencies
+        cabal v2-configure --project-file=cabal.project.ci --enable-benchmarks --enable-tests --write-ghc-environment-files=always -j2
+        cabal v2-build     --project-file=cabal.project.ci --only-dependencies all
 
     - name: Build & test
       run: |
-        cabal v2-build
-        cabal v2-run test
-        cabal v2-run examples
-        cabal v2-haddock
-        cabal v2-sdist
-        cabal check
+        cabal v2-build   --project-file=cabal.project.ci
+        cabal v2-run     --project-file=cabal.project.ci test
+        cabal v2-run     --project-file=cabal.project.ci examples
+        cabal v2-haddock --project-file=cabal.project.ci
+        cabal v2-sdist   --project-file=cabal.project.ci
+        cabal check      --project-file=cabal.project.ci
```

Comment:
Having a CI-specific project file allows me to not have to use `-Werror` when building locally, without CI running into the problem where `--ghc-options=-Werror` gets applied to dependencies as well.